File: src/leetcode/python/easy/ConvertANumberToHexadecimal_405.py
class ConvertANumberToHexadecimal_405(object):
    def toHex(self, num):
        """
        :type num: int
        :rtype: str
        """
        if num == 0:
            return '0'
        res, feq = "", 0
        while feq < 8 and num:
            res = "0123456789abcdef"[num & 15] + res
            num >>= 4
            feq += 1
        return res

File: src/leetcode/java/ConvertANumberToHexadecimal_405.java
package leetcode.java;

public class ConvertANumberToHexadecimal_405 {
  final char[] hexMap = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };

  public String toHex(int num) {
    String result = "";
    while (num != 0) {
      result = hexMap[(num & 15)] + result;
      num >>>= 4;
    }
    return result;
  }
}
